tools/Common/IceRpc.BuildTelemetry/TelemetryTask.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) ZeroC, Inc.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace IceRpc.BuildTelemetry;
+
+/// <summary>
+/// A custom MSBuild task to run telemetry commands.
+/// </summary>
+public class TelemetryTask : ToolTask
+{
+    /// <summary>
+    /// Gets or sets the compilation hash.
+    /// </summary>
+    [Required]
+    public string CompilationHash { get; set; } = "";
+
+    /// <summary>
+    /// Gets or sets the source.
+    /// </summary>
+    [Required]
+    public string Source { get; set; } = "";
```

Comment:
I'm still not a super big fan of the term "source", it's just so vague, and there's no further explanation about it.
Is this the IP address? A full copy of their source code? I know it has something to do with "Slice" vs "Protobuf" but that's it.

Reply:
This property will go away when we split the Slice definition into a Slice struct and Protobuf struct so I would prefer to just leave it as is for now!